ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1026,7 +1026,7 @@ private struct ZipState<Left, Right> {
 	var completed: (left: Bool, right: Bool) = (false, false)
 
 	var isFinished: Bool {
-		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty)
+		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty) || (completed.left && completed.right)
```

Comment:
@mdiep It doesn't prematurely end the signal. The conditions here only determines whether it should end the signal after sending a tuple (if any).